### PR TITLE
Optional Burn in for the models

### PR DESCRIPTION
**Pull request by Polkas**
- [x] add man docs (make sure roxygen2 is not used)
- [x] add unit tests
- [x] add reference to stackoverflow answer (added here but eventually added to PR from our master to original master)
- [x] provide examples for the gemtc package author  (added here but eventually added to PR from our master to original master)
- [x] update the NEWS.md
- [x] update the DESCRIPTION file

### CODE REVIEW

**Comment by frostwix** [comment]
We want to remember to exclude the .gitignore from the PR to the original master

**Comment by frostwix** [comment]
mtc.run is included in the following tests: helper-regress.R, helper-validate.R, test-regress-models.R

File: gemtc/R/mtc.run.R [hunk 1]
```diff
@@ -64,6 +64,9 @@ mtc.sample <- function(model, n.adapt=n.adapt, n.iter=n.iter, thin=thin) {
     inits=syntax[['inits']],
     n.chains=model[['n.chain']],
     n.adapt=n.adapt)
+  
+  update(jags, n.iter=n.burnin)
```
Comment (frostwix) [inline gemtc/R/mtc.run.R]: `rjags::update(jags, n.iter=n.burnin)`

**Comment by Polkas** [comment]
proof that the solution is backward compatible so the all runs are the same. This is what we should test in tests

**Comment by Polkas** [comment]
URL to share in the final PR: https://stackoverflow.com/a/38875637/5442527 that nadapt != nburnin

**Comment by Polkas** [comment]
We want to make a proposition to update the `mtc.run` function as it is not supporting a burn-in phase currently.
The burn-in phase is often required by national agencies for all mcmc runs, and even for conjugates.
We make the  `mtc.run` update as minimalistic as posible and backward compatible. 
That is why for backward compatiblity the NEW ARGUMENT `n.burnin` is added as the last one, so `mtc.run`  will works even if somebody used it without argument names `mtc.run(value, value)` instead of  `mtc.run(arg1 = value, arg2 = value)`.
Please give us your thoughts on the update.

Adaptation and burnin are not the same and currently only the former can be set in `mtc.run`.
Please read more about it here https://stackoverflow.com/a/78807748/5442527  ([example where adaptation is skipped as all models are conjugates](https://stackoverflow.com/q/78536214/5442527)) and here https://stackoverflow.com/a/38875637/5442527. Summing up it is important to recognize that burn-in and adaptation refer to two different things and are implemented independently in JAGS/rjags. The adaptation will be run only if at least one parameter in the model requires it, whereas burn-in is run always.

Our update fix the https://github.com/gertvv/gemtc/issues/78 issue.

We did not add any unit tests as the current tests structure will demand a lot of copy pasting. Please feel free to propose something.

Our update is backward compatible and results from mtc.run are the same.
The default value for n.burnin is 0.
Proof that the results from mtc.run are the same.

```
remotes::install_github("gertvv/gemtc/gemtc@master", force = TRUE)

library(gemtc)

set.seed(42)
model <- mtc.model(parkinson)
# By default, the model will have 4 chains - generate a seed for each
seeds <- sample.int(4, n = .Machine$integer.max)
# Apply JAGS RNG settings to each chain
model$inits <- mapply(c, model$inits, list(
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[1]),
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[2]),
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[3]),
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[4])), SIMPLIFY=FALSE)

results <- mtc.run(model, thin=1)

saveRDS(results, "mtc_run_before.RDS")

remotes::install_github("frostwix/gemtc/gemtc@burnin", force = TRUE)

library(gemtc)

set.seed(42)
model <- mtc.model(parkinson)
# By default, the model will have 4 chains - generate a seed for each
seeds <- sample.int(4, n = .Machine$integer.max)
# Apply JAGS RNG settings to each chain
model$inits <- mapply(c, model$inits, list(
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[1]),
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[2]),
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[3]),
  list(.RNG.name="base::Mersenne-Twister", .RNG.seed=seeds[4])), SIMPLIFY=FALSE)

results <- mtc.run(model, thin=1)

saveRDS(results, "mtc_run_after.RDS")

identical(readRDS("mtc_run_before.RDS")$samples, readRDS("mtc_run_after.RDS")$samples)
# TRUE
identical(readRDS("mtc_run_before.RDS")$deviance, readRDS("mtc_run_after.RDS")$deviance)
# TRUE
```

**Comment by Polkas** [comment]
We should add a link to the regulatory which shows that it requires to run burnin always

File: gemtc/R/mtc.run.R [hunk 2]
```diff
@@ -19,7 +19,12 @@ mtc.model.run <- function(network, type, ...) {
 
 # If is.na(sampler), a sampler will be chosen based on availability, in this order:
 # JAGS, BUGS. When the sampler is BUGS, BRugs or R2WinBUGS will be used.
-mtc.run <- function(model, sampler=NA, n.adapt=5000, n.iter=20000, thin=1) {
+mtc.run <- function(model, sampler=NA, n.adapt=5000, n.burnin=0, n.iter=20000, thin=1) {
```
Comment (Polkas) [inline gemtc/R/mtc.run.R]:
```suggestion
mtc.run <- function(model, sampler=NA, n.adapt=5000, n.iter=20000, thin=1, n.burnin=0) {
```

File: .gitignore [hunk 3]
```diff
@@ -4,3 +4,5 @@ gemtc.Rcheck
 gemtc_*.tar.gz
 *.so
 *.o
+.Rproj.user
+gemtc.Rproj
```
Comment (Polkas) [inline .gitignore]:
```suggestion
```

File: gemtc/NEWS.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+gemtc 1.0-2.9000
+===========
+
+Released 2023-06-21
```
Comment (Polkas) [inline gemtc/NEWS.md]:
```suggestion
Released 2024-XX-XX
```

File: gemtc/R/mtc.run.R [hunk 5]
```diff
@@ -19,7 +19,12 @@ mtc.model.run <- function(network, type, ...) {
...
+  stopifnot(is.numeric(n.adapt))
+  stopifnot(is.numeric(n.burnin))
+  stopifnot(is.numeric(n.iter))
+  stopifnot(is.numeric(thin))
```
Comment (Polkas) [inline gemtc/R/mtc.run.R]:
```suggestion
```

File: gemtc/R/mtc.run.R [hunk 6]
```diff
@@ -44,7 +49,7 @@ mtc.build.syntaxModel <- function(model) {
   )
 }
 
-mtc.sample <- function(model, n.adapt=n.adapt, n.iter=n.iter, thin=thin) {
+mtc.sample <- function(model, n.adapt, n.burnin, n.iter, thin) {
```
Comment (Polkas) [inline gemtc/R/mtc.run.R]:
```suggestion
mtc.sample <- function(model, n.adapt, n.iter, thin, n.burnin) {
```

File: gemtc/man/mtc.run.Rd [hunk 7]
```diff
@@ -14,7 +14,7 @@ The function \code{mtc.run} is used to generate samples from a object of type \c
 The resulting \code{mtc.result} object can be coerced to an \code{mcmc.list} for further analysis of the dataset using the \code{coda} package.
 }
 \usage{
-mtc.run(model, sampler = NA, n.adapt = 5000, n.iter = 20000, thin = 1)
+mtc.run(model, sampler = NA, n.adapt = 5000, n.burnin = 0, n.iter = 20000, thin = 1)
```
Comment (Polkas) [inline gemtc/man/mtc.run.Rd]:
```suggestion
mtc.run(model, sampler = NA, n.adapt = 5000, n.iter = 20000, thin = 1, n.burnin = 0)
```

File: gemtc/man/mtc.run.Rd [hunk 8]
```diff
@@ -26,6 +26,7 @@ mtc.run(model, sampler = NA, n.adapt = 5000, n.iter = 20000, thin = 1)
   \item{model}{An object of S3 class \code{mtc.model} describing a network meta-analysis model.}
   \item{sampler}{Deprecated: gemtc now only supports the JAGS sampler. Specifying a sampler will result in a warning or error. This argument will be removed in future versions.}
   \item{n.adapt}{Amount of adaptation (or tuning) iterations.}
+  \item{n.burnin}{Amount of burn-in iterations. Has to be positive number.}
   \item{n.iter}{Amount of simulation iterations.}
   \item{thin}{Thinning factor.}
```
Comment (Polkas) [inline gemtc/man/mtc.run.Rd]:
```suggestion
  \item{n.iter}{Amount of simulation iterations.}
  \item{thin}{Thinning factor.}
  \item{n.burnin}{Amount of burn-in iterations. Has to be positive number.}
```

File: gemtc/R/mtc.run.R [hunk 9]
```diff
@@ -28,7 +33,7 @@ mtc.run <- function(model, sampler=NA, n.adapt=5000, n.iter=20000, thin=1) {
     }
   }
 
-  result <- mtc.sample(model, n.adapt=n.adapt, n.iter=n.iter, thin=thin)
+  result <- mtc.sample(model, n.adapt=n.adapt, n.burnin = n.burnin, n.iter=n.iter, thin=thin)
```
Comment (Polkas) [inline gemtc/R/mtc.run.R]:
```suggestion
  result <- mtc.sample(model, n.adapt=n.adapt, n.iter=n.iter, thin=thin, n.burnin = n.burnin)
```

**Pull request merged**